src/Reshuffle.ts

```diff
@@ -6,6 +6,24 @@ import { BaseConnector, BaseHttpConnector, EventConfiguration } from 'reshuffle-
 import { createLogger } from './Logger'
 import { Logger, LoggerOptions } from 'winston'
 import http from 'http'
+import fs from 'fs'
+
+function getReshufflePackagesList(directory = __dirname): Record<string, string> {
+  const filename = 'package.json'
+  const dirSplit = directory.split('/')
+  const dir = dirSplit.slice(0, dirSplit.length - 1).join('/')
+
+  try {
+    if (fs.readdirSync(dir).includes(filename)) {
```

Comment:
readdirSync and readFileSync will block the entire event loop in NodeJS, which is not a good idea. Use "import { promises as fs } from 'fs'" and them "await readdir" and "await readFile"

Reply:
very good point, fixed